contributors/design-proposals/scheduling/scheduling-framework.md

```diff
@@ -1,436 +1,620 @@
+
+1.  The number of extension points are limited: "Filter" extenders are called
+    after default predicate functions. "Prioritize" extenders are called after
+    default priority functions. "Preempt" extenders are called after running
+    default preemption mechanism. "Bind" verb of the extenders are used to bind
+    a Pod. Only one of the extenders can be a binding extender, and that
+    extender performs binding instead of the scheduler. Extenders cannot be
+    invoked at other points, for example, they cannot be called before running
+    predicate functions.
+1.  Every call to the extenders involves marshaling and unmarshalling JSON.
+    Calling a webhook (HTTP request) is also slower than calling native
+    functions.
+1.  It is hard to inform an extender that scheduler has aborted scheduling of a
+    Pod. For example, if an extender provisions a cluster resource and scheduler
+    contacts the extender and asks it to provision an instance of the resource
+    for the pod being scheduled and then scheduler faces errors scheduling the
+    pod and decides to abort the scheduling, it will be hard to communicate the
+    error with the extender and ask it to undo the provisioning of the resource.
+1.  Since current extenders run as a separate process, they cannot use
+    scheduler's cache. They must either build their own cache from the API
+    server or process only the information they receive from the default
+    scheduler.
 
 The above limitations hinder building high performance and versatile scheduler
-extensions. We would ideally like to have an extension mechanism that is fast
-enough to allow keeping a bare minimum logic in the scheduler core and convert
-many of the existing features of default scheduler, such as predicate and
-priority functions and preemption into plugins. Such plugins will be compiled
-with the scheduler. We would also like to provide an extension mechanism that do
-not need recompilation of scheduler. The expected performance of such plugins is
-lower than in-process plugins. Such out-of-process plugins should be used in
-cases where quick invocation of the plugin is not a constraint.
+features. We would ideally like to have an extension mechanism that is fast
+enough to allow existing features to be converted into plugins, such as
+predicate and priority functions. Such plugins will be compiled into the
+scheduler binary.
 
 # OVERVIEW
 
-Scheduler v2 allows both built-in and out-of-process extenders. This new
-architecture is a scheduling framework that exposes several extension points
-during a scheduling cycle. Scheduler plugins can register to run at one or more
-extension points.
+The Scheduling Framework defines new extension points and Go APIs in the
+Kubernetes Scheduler for use by "plugins". Plugins add scheduling behaviors to
+the scheduler, and are included at compile time. The scheduler's ComponentConfig
+will allow plugins to be enabled, disabled, and reordered.
```

Comment:
So the plugins live in-tree ?

Reply:
Once again, our answer is "both". I've added clarification.